src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenRefLocalTests.cs

```diff
@@ -1697,5 +1698,31 @@ static void M()
             Assert.Null(model.GetSymbolInfo(refInt).Symbol);
             Assert.Null(model.GetTypeInfo(refInt).Type);
         }
+
+        [WorkItem(17395, "https://github.com/dotnet/roslyn/issues/17453")]
+        [Fact]
+        public void Regression17395()
+        {
+            var source = @"
+using System;
+
+public class C 
+{            
+    public void F()
+    {
+        ref int[] a = ref {1,2,3};
+        Console.WriteLine(a[0]);
+    }        
+}
+";
+
+            var c = CreateCompilationWithMscorlib45AndCSruntime(source);
+
+            c.VerifyEmitDiagnostics(
+                // (8,27): error CS1510: A ref or out value must be an assignable variable
+                //         ref int[] a = ref {1,2,3};
+                Diagnostic(ErrorCode.ERR_RefLvalueExpected, "{1,2,3}").WithLocation(8, 27)
+            );
```

Comment:
NO, regular EmitDiagnostics is enough. I think, this should be changed.

---
In reply to: [103610967](https://github.com/dotnet/roslyn/pull/17477#discussion_r103610967) [](ancestors = 103610967)